pkg/common/cns-lib/volume/manager.go

```diff
@@ -973,9 +973,21 @@ func (m *defaultManager) DetachVolume(ctx context.Context, vm *cnsvsphere.Virtua
 		volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
 		if volumeOperationRes.Fault != nil {
 			faultType = ExtractFaultTypeFromVolumeResponseResult(ctx, volumeOperationRes)
+
```

Comment:
I am a bit confused. Perhaps there is some history to these changes that I am unaware of. I see similar code in https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/f21ff747b18d8198219ab874853dbc2e68c0a12e/pkg/common/cns-lib/volume/manager.go#L916 , just a couple of lines above, does this mean that previously `DetachVolume` would return a fault on API invocation rather than setting the fault in the task?

Reply:
There is a comment from Chethan:
I am assuming the need for this change is due to namespace deletion and detach volume getting invoked concurrently(or they are invoked in a very tiny time window) and CNS task itself did not fail immediately, but a fault was returned from the task output after a CNS Detach invocation.

I think it answers your query.